api4/resolver.go

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v6/app"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/web"
+)
+
+// cursorPrefix is used to categorize objects
+// sent in a cursor. The type is prepended
+// to the string with a - to find which
+// object the id belongs to.
+//
+// And after the type is extracted, object
+// specific logic can be applied to extract the id.
+type cursorPrefix string
+
+const (
+	channelMemberCursorPrefix cursorPrefix = "channelMember"
+	channelCursorPrefix       cursorPrefix = "channel"
+)
+
+type resolver struct {
+}
+
+// match with api4.getChannelsForTeamForUser
+func (r *resolver) Channels(ctx context.Context, args struct {
+	TeamID         string
+	UserID         string
+	IncludeDeleted bool
+	LastDeleteAt   float64
+	LastUpdateAt   float64
+	First          int32
+	After          string
+}) ([]*channel, error) {
+	c, err := getCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if args.UserID == model.Me {
+		args.UserID = c.AppContext.Session().UserId
+	}
+
+	if !c.App.SessionHasPermissionToUser(*c.AppContext.Session(), args.UserID) {
+		c.SetPermissionError(model.PermissionEditOtherUsers)
+		return nil, c.Err
+	}
+
+	if args.TeamID != "" && !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), args.TeamID, model.PermissionViewTeam) {
+		c.SetPermissionError(model.PermissionViewTeam)
+		return nil, c.Err
+	}
+
+	limit := int(args.First)
+	// ensure args.First limit
+	if limit == 0 {
+		limit = web.PerPageDefault
+	} else if limit > web.PerPageMaximum {
+		return nil, fmt.Errorf("first parameter %d higher than allowed maximum of %d", limit, web.PerPageMaximum)
+	}
+
+	// ensure args.After format
+	var afterChannel string
+	var ok bool
+	if args.After != "" {
+		afterChannel, ok = parseChannelCursor(args.After)
+		if !ok {
+			return nil, fmt.Errorf("after cursor not in the correct format: %s", args.After)
+		}
```

Comment:
Hmm .. I think this is okay. Does it look scattered because of the empty string check? I'm not overly opinionated either ways, but wanted to check if there's any other reason the logic seems scattered to you.

Reply:
Yeah, I though the empty string check can be done in `parseChannelCursor`. 0/5 though

Reply:
Ok, keeping it as is for now. Let me know if you feel otherwise.